webpack/assets/javascripts/foreman_tools.test.js

```diff
@@ -57,3 +57,51 @@ describe('deprecate', () => {
     expect(console.warn).toHaveBeenCalledWith('DEPRECATION WARNING: you are using deprecated oldtest, it will be removed in Foreman 1.42. Use tfm.tools.newtest instead.');
   });
 });
+
+/* eslint-disable max-statements */
+describe('initTypeAheadSelect', () => {
```

Comment:
I'm not sure I completely understand this test, but it looks like you are verifying the internals of select2 work correctly? Would it be possible to just mock `$.ajax` so it returns results in the correct format when called, then check that the field has indeed been initialized correctly after the call (i.e. the correct options are there)? That way if something in the internals changes we wouldn't care unless it broke the functionality.